src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -206,12 +196,16 @@ public ListenerToken AddChangeListener(EventHandler<ReplicatorStatusChangedEvent
         }
 
         /// <summary>
-        /// Removes a previously added change listener via its <see cref="ListenerToken"/>
+        /// Removes a previously added change listener via its <see cref="ListenerToken"/> and/or
+        /// Removes a previously added documents ended listener via its <see cref="ListenerToken"/>
         /// </summary>
-        /// <param name="token">The token received from <see cref="AddChangeListener(TaskScheduler, EventHandler{ReplicatorStatusChangedEventArgs})"/></param>
+        /// <param name="token">The token received from <see cref="AddChangeListener(TaskScheduler, EventHandler{ReplicatorStatusChangedEventArgs})"/>
+        /// and/or The token received from <see cref="AddReplicationListener(TaskScheduler, EventHandler{DocumentReplicatedEventArgs})"/></param>
         public void RemoveChangeListener(ListenerToken token)
         {
             _statusChanged.Remove(token);
```

Comment:
I'm not sure this will be ok with the wrong token type.  You should check the type and perform the correct removal.

Reply:
It works as is. The token passed into the Remove method, it returns 0 if no token with the correct id to remove or if the token with the correct id is removed. If not the correct id Token pass into the method and token exists, Remove method will return 1.

Reply:
Ok, I was concerned about the token being `null` inside but if you have tested that this combined method works then it should be fine.